core/src/main/java/com/novoda/noplayer/exoplayer/forwarder/OnCompletionForwarder.java

```diff
@@ -0,0 +1,55 @@
+package com.novoda.noplayer.exoplayer.forwarder;
+
+import com.google.android.exoplayer2.ExoPlaybackException;
+import com.google.android.exoplayer2.ExoPlayer;
+import com.google.android.exoplayer2.PlaybackParameters;
+import com.google.android.exoplayer2.Timeline;
+import com.google.android.exoplayer2.source.TrackGroupArray;
+import com.google.android.exoplayer2.trackselection.TrackSelectionArray;
+import com.novoda.noplayer.listeners.CompletionListeners;
+
+class OnCompletionForwarder implements ExoPlayer.EventListener {
+
+    private final CompletionListeners completionListeners;
+
+    OnCompletionForwarder(CompletionListeners completionListeners) {
+        this.completionListeners = completionListeners;
+    }
+
+    @Override
+    public void onPlayerStateChanged(boolean playWhenReady, int playbackState) {
+        if (playbackState == ExoPlayer.STATE_ENDED) {
+            completionListeners.onCompletion();
+        }
+    }
+
+    @Override
+    public void onTimelineChanged(Timeline timeline, Object manifest) {
+        //TODO should we send ?
+    }
+
+    @Override
+    public void onTracksChanged(TrackGroupArray trackGroups, TrackSelectionArray trackSelections) {
+        //TODO should we send ?
+    }
+
+    @Override
+    public void onLoadingChanged(boolean isLoading) {
+        //TODO should we send ?
+    }
+
+    @Override
+    public void onPlayerError(ExoPlaybackException error) {
+        //Sent by ErrorForwarder
```

Comment:
Ah. This particular event is handled by the `errorForwarder` and should not be handled by this `Forwarder`. This is because it would be a bit strange to notify the `no-player` complete listeners of an error, when that is handled by the `no-player` error listeners. 

This is a small problem with using the interfaces we are interested in from `ExoPlayer` it leads to explicit no-ops in the specific implementations.